addons/hr_payroll/controllers/main.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import io
+import re
+
+from PyPDF2 import  PdfFileReader, PdfFileWriter
+
+from odoo.http import request, route, Controller
+from odoo.tools.safe_eval import safe_eval
+
+
+class HrPayroll(Controller):
+
+    @route(["/hr/payroll/print_report"], type='http', auth='user')
+    def get_payroll_report_print(self, list_ids='', **post):
+        if not request.env.user.has_group('hr_payroll.group_hr_payroll_user'):
+            return request.not_found()
+
+        if not list_ids:
+            return request.not_found()
+
+        if re.search("[^0-9|,]", list_ids):
+            return request.not_found()
+
+        ids = [int(s) for s in list_ids.split(',')]
+        payslips = request.env['hr.payslip'].search([('id', 'in', ids)])
+
+        if not payslips:
```

Comment:
Not usefull, you'll always have something, even if you type `request.env['hr.payslip'].browse('test')`